include/eve/module/core/regular/impl/simd_cast.hpp

```diff
@@ -0,0 +1,38 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/function/bit_cast.hpp>
+#include <eve/detail/function/combine.hpp>
+
+namespace eve::detail
+{
+template<callable_options O, typename T, typename Target>
+EVE_FORCEINLINE constexpr Target
+simd_cast_(EVE_REQUIRES(cpu_), O const&, T x, as<Target> tgt) noexcept
+{
+  if constexpr( product_simd_value<T> )
+  {
+    auto make_as       = []<typename M>(M) { return as<M> {}; };
+    auto lifted_target = kumi::result::map_t<decltype(make_as), Target> {};
```

Comment:
sidenote: should kumi provides a lift() function for this kind of scenario ?

Reply:
I don't like that I had to create a lambda.  But I dunno - this is some mpl style shenanigans.

Reply:
OK, will make a KUMI issue thoguh, it looks like a valid use case